src/basic/path-util.c

```diff
@@ -794,6 +794,89 @@ char* dirname_malloc(const char *path) {
         return dir2;
 }
 
+static const char *drop_slash_or_dot(const char *p) {
+        for (; !isempty(p); p++) {
+                if (*p == '/')
+                        continue;
+                if (startswith(p, "./")) {
+                        p++;
+                        continue;
+                }
+                break;
+        }
+        return p;
+}
+
+int path_get_first_component(const char **p, bool accept_dot_dot, const char **ret) {
```

Comment:
maybe call this "path_find_first_component()"? 

i.e. "get" sounds you'd actually return something you actively acquire. but this here is just about finding the position of two concepts in the path